### PR TITLE
refactor: change pre-decrement/increment to post

### PR DESCRIPTION
**Pull request by lewis6991**
Ran the commands:

```bash
sed -i 's/^\(\s*\)--\(\w\+\);$/\1\2--;/g' **/*.c
sed -i 's/^\(\s*\)++\(\w\+\);$/\1\2++;/g' **/*.c
```

And reviewed the output by hand.

We can add this to `.git-blame-ignore-revs` when it is merged.

### CODE REVIEW

**Comment by zeertzjq** [comment]
Should xdiff be changed?

**Comment by lewis6991** [comment]
Actually we don't even run clint on xdiff, so I'll just back off the changes.

**Comment by clason** [comment]
(I don't think `libxdiff` is on github...)

**Comment by lewis6991** [comment]
> (I don't think libxdiff is on github...)

We don't use `libxdiff`, we use git's version of xdiff: https://github.com/git/git/tree/master/xdiff

Git's version is quite a bit leaner than `libxdiff`.

File: src/nvim/edit.c [hunk 1]
```diff
@@ -3794,12 +3794,9 @@ bool in_cinkeys(int keytyped, int when, bool line_is_empty)
...
         look++;
       }
-    }
-    /*
-     * Is it a word: "=word"?
-     */
-    else if (*look == '=' && look[1] != ',' && look[1] != NUL) {
-      ++look;
+    } else if (*look == '=' && look[1] != ',' && look[1] != NUL) {
+      // Is it a word: "=word"?
```
Comment (zeertzjq) [inline src/nvim/edit.c:3798]: This is clearer about what the comment refers to
```suggestion
      // Is it a word: "=word"?
    } else if (*look == '=' && look[1] != ',' && look[1] != NUL) {
```

Comment (lewis6991) [reply src/nvim/edit.c:3798]: I kind of disagree on this one, and the suggestion here goes against the convention we've been applying when fixing these lint errors.

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**